xbatcher/testing.py

```diff
@@ -183,7 +183,7 @@ def validate_batch_dimensions(
 
     # Check the names and lengths of the dimensions are equal
     TestCase().assertDictEqual(
-        expected_dims, batch.sizes.mapping, msg="Dimension names and/or lengths differ"
+        expected_dims, dict(batch.sizes), msg="Dimension names and/or lengths differ"
```

Comment:
Applying this patch to fix `error: "Mapping[Hashable, int]" has no attribute "mapping"  [attr-defined]`, due to what might be an incorrect type hint in upstream xarray. The `batch.sizes` variable should probably have a type hint of [`xarray.core.utils.Frozen`](https://github.com/pydata/xarray/blob/c47f35db4e73507b4ac729ec23b0faf8ceb5ace1/xarray/core/utils.py#L448-L472) instead?